### ISSUE TITLE
False Negative on Webseclab XSS test for Referer Header echo (/xss/reflect/refer1)

### ISSUE THREAD

**Issue by dmitris** [Arachni/arachni#583]
Consider the Webseclab test echoing unfiltered value of the incoming HTTP Referer header in the response:
```
curl -H 'Referer: bar<script>alert(123)</script>' http://127.0.0.1:8080/xss/reflect/refer1
[...]
You have arrived following the referal of the site: bar<script>alert(123)</script> (unescaped version: bar<script>alert(123)</script>)
```

Arachni currently does not flag XSS or Referer Header injection which can lead to missing related cases.
To exploit this, the attacker would ned to lure the user to an intermediate site with the attack payload in the URL (possibly at the end of a long string of cgi parameters) and then get the user to click on a link to the target site.  It is more involved that a straight reflected XSS but in general exploits of the Referer header injection are relatively easy (especially compared to other HTTP headers).  Those cases do come up in practice when developers decide to use the value of Referer header for certain purposes (for example, to fill in the "Back" URL, etc.)  It would be great to add checks for such cases.

Arachni scan results from the Web UI:
![screen shot 2015-05-12 at 14 39 20](https://cloud.githubusercontent.com/assets/31205/7587182/c2700aac-f8b4-11e4-8425-75a911b043c0.png)

References:
http://www.gremwell.com/exploiting_xss_in_referer_header

**Comment by Zapotek** [comment]
Silly question but did you enable auditing headers?

**Comment by dmitris** [comment]
No, not silly at all.  In fact, I did not have auditing headers turned on - did not realize I have to edit the Cross-Site Scripting profile for this check to become active.  My preference would be to have it turned on by default in the XSS profile but of course it can be approached differently.

Closing the issue, thanks.

![screen shot 2015-05-12 at 9 24 40 pm](https://cloud.githubusercontent.com/assets/31205/7596464/2a7b2e7c-f8ee-11e4-8489-4da83e6b538f.png)
